File: University_Documentations/GuessTheActressV1/preview_files/MainActivity.java
package com.example.guesstheactressv1;


import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentTransaction;

import android.os.Bundle;
public class MainActivity extends AppCompatActivity {
    CelebrityFetch fetch;

    @Override
    public void onBackPressed() {
        fetch.cancel(true);
        finishAffinity();
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        FragmentTransaction fragmentTransaction = getSupportFragmentManager().beginTransaction();
        fragmentTransaction.replace(R.id.frameLayoutMain, new LoadingFragment(this));
        fragmentTransaction.commit();

        fetch = new CelebrityFetch(this);
        fetch.execute();
    }
}

File: LearningFragment/app/src/main/java/com/example/learningfragment/MainActivity.kt
package com.example.learningfragment

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun switchFragment1(view: View){
        val fragmentManager = supportFragmentManager
        val fragmentTransaction = fragmentManager.beginTransaction()
        val firstFragment = BlankFragment()
        fragmentTransaction.replace(R.id.frameLayout,firstFragment)
        fragmentTransaction.commit()

    }

    fun switchFragment2(view: View){
        val fragmentManager = supportFragmentManager
        val fragmentTransaction = fragmentManager.beginTransaction()
        val secondFragment = BlankFragment2()
        fragmentTransaction.replace(R.id.frameLayout,secondFragment)
        fragmentTransaction.commit()
    }
}